forge-core/src/main/kotlin/io/github/hellocuriosity/ModelForge.kt

```diff
@@ -89,6 +90,26 @@ open class ModelForge {
         return list
     }
 
+    /**
+     *  Creates an automatically generated model map with
+     *  a specified size.
+     *
+     *  @param type ParameterizedType to instantiate
+     *  @param size Int number of models to create
+     *
+     *  @return Instance of clazz
+     */
+    private fun buildMap(type: ParameterizedType, size: Int): Map<Any, Any> {
+        val key = type.actualTypeArguments[0] as Class<*>
+        val value = type.actualTypeArguments[1] as Class<*>
+
+        val map = mutableMapOf<Any, Any>()
+        while (map.size < size) {
+            map[build(key)] = build(value)
+        }
+        return map
```

Comment:
@nxtstep what do you think of this? I'm not sure if I'm completely convinced myself 🤔  I originally had this:

```suggestion
        return (0 until size).map {
            build(key) to build(value)
        }.toMap()
```

However, this was quite unstable because more often than not the key would be a duplication and then the `map` would have a size of 8 or 9 instead of 10 for example 🙈

Reply:
Yeah, the `0 until hero` solution won't work if you must reach `size` items per se.
(unbounded) while loops always have infinite potential 🎉  consider always adding an `iKillInfinite` counter/switch to break out when in becomes unreasonable

Reply:
> while loops always have infinite potential 🎉

Yes they sure do 😅 I was also contemplating a kill switch of some type